validate-infra/action.yml

```diff
@@ -0,0 +1,136 @@
+name: Validate Infrastructure
+description: Run terraform plan with detailed exit code to validate infrastructure state
+
+inputs:
+  azure-client-id:
+    description: Azure service principal or managed identity client ID when using OIDC
+    required: false
+    default: ""
+  azure-subscription-id:
+    description: Azure service principal or managed identity subscription ID when using OIDC
+    required: false
+    default: ""
+  azure-tenant-id:
+    description: Azure service principal or managed identity tenant ID when using OIDC
+    required: false
+    default: ""
+  environment:
+    description: Environment to validate (test, platform-test, production)
+    required: true
+  terraform-base:
+    description: Path to the terraform files
+    required: false
+    default: "cluster/terraform_aks_cluster"
+  slack-webhook:
+    description: Slack webhook URL for notifications
+    required: false
+    default: ""
+
+outputs:
+  drift_detected:
+    description: Whether infrastructure drift was detected
+    value: ${{ steps.plan.outputs.drift_detected }}
+
+runs:
+  using: composite
+
+  steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+
+    - name: Switch to latest main
+      shell: bash
+      run: |
+        # Fetch the latest main branch
+        git fetch origin main
+        # Get and checkout the latest main SHA
+        MAIN_SHA=$(git rev-parse origin/main)
```

Comment:
we don't know it's main, which is why we should run as per https://github.com/DFE-Digital/github-actions/blob/master/build-docker-image/action.yml#L80 
plus we then need to set IMAGE_TAG and DOCKER_IMAGE_TAG as these are what we use in Makefile deploy